### ISSUE TITLE
Does not record to disc

### ISSUE THREAD

**Issue by DAN13L83** [Jo2003/cd2netmd_gui#10]
2023-09-16 17:57:30.170|INFO|cnetmd.cpp:39|CNetMD::CNetMD(QObject *)|Log File / Stream "/private/var/folders/t6/s177fksd1lg0sl6h_9lty01r0000gn/T/cd2netmd_transfer_log.tmp" created.
2023-09-16 17:57:35.300|INFO|cjacktheripper.cpp:537|int CJackTheRipper::cddbReqString()|First track:  1 Last track:  21
2023-09-16 17:57:35.300|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 0 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 1 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 2 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 3 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 4 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 5 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 6 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 7 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 8 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 9 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 10 is named "<untitled>"
2023-09-16 17:57:35.301|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 11 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 12 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 13 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 14 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 15 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 16 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 17 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 18 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 19 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 20 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cjacktheripper.cpp:196|int CJackTheRipper::parseCDText(cdtext_t *, track_t, QString &)|Title 21 is named "<untitled>"
2023-09-16 17:57:35.302|INFO|cdrutil.cpp:37|int CDRUtil::start()|drutil ("cdtext")
2023-09-16 17:57:35.522|INFO|cjacktheripper.cpp:479|int CJackTheRipper::cddbRequest()|No CD-Text, do CDDB request:  "2912a415+21+0+23980+42755+62395+78635+96102+111615+128670+146300+158055+177565+196190+210905+227187+249180+266605+281597+289360+300682+323120+338242+4772"
2023-09-16 17:57:35.524|INFO|ccddb.cpp:51|int CCDDB::getEntries(const QString &, c2n::AudioTracks &)|Request entries:  QUrl("https://gnudb.gnudb.org:443/~cddb/cddb.cgi?cmd=cddb+query+2912a415+21+0+23980+42755+62395+78635+96102+111615+128670+146300+158055+177565+196190+210905+227187+249180+266605+281597+289360+300682+323120+338242+4772&hello=me@you.org+localhost+MyRipper+0.1.0&proto=6")
2023-09-16 17:57:39.923|INFO|mainwindow.cpp:173|void MainWindow::catchCDDBEntries(QStringList)|Request CDDB match:  "misc+0912a315"
2023-09-16 17:57:39.923|INFO|ccddb.cpp:67|int CCDDB::getEntry(const QString &)|Request matching entry:  QUrl("https://gnudb.gnudb.org:443/~cddb/cddb.cgi?cmd=cddb+read+misc+0912a315&hello=me@you.org+localhost+MyRipper+0.1.0&proto=6")
2023-09-16 17:57:40.191|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 17:57:40.191|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 17:57:48.230|INFO|cnetmd.cpp:132|int CNetMD::getDiscInfo()|getting MD disc / device info ...
2023-09-16 17:57:48.438|INFO|cnetmd.cpp:256|int CNetMD::getDiscInfo()|{
    "device": "Sony MZ-NE410/DN430/NF520",
    "disc_flags": "0x10",
    "groups": [
    ],
    "otf_enc": 0,
    "sp_upload": 1,
    "t_free": 4859,
    "t_total": 4859,
    "t_used": 0,
    "title": "<untitled>",
    "toc_manip": 1,
    "tracks": [
    ],
    "trk_count": 0
}
2023-09-16 17:57:48.446|INFO|cnetmd.cpp:522|void CNetMD::procEnded(bool, int)|
2023-09-16 17:57:48|INFO|initDevice():239: Product name: Net MD Walkman, serial number:
2023-09-16 17:57:48.447|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: false ret: 0
2023-09-16 17:57:55.177|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 17:57:55.177|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 17:57:55.180|INFO|cjacktheripper.cpp:117|int CJackTheRipper::extractTrack(int, const QString &, bool)|Extract track -1 to /private/var/folders/t6/s177fksd1lg0sl6h_9lty01r0000gn/T/cd2netmd.Qe9P9c.tmp ...
2023-09-16 17:57:55.180|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: true ret: 0
2023-09-16 18:04:06.161|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: true ret: 0
2023-09-16 18:04:06.164|INFO|cnetmd.cpp:272|int CNetMD::writeTrack(const CNetMD::NetMDCmd &, const QString &, const QString &)|send track: "DAO All in One!" file: "/private/var/folders/t6/s177fksd1lg0sl6h_9lty01r0000gn/T/cd2netmd.Qe9P9c.tmp" to "Sony MZ-NE410/DN430/NF520"
2023-09-16 18:05:42.999|CRITICAL|cnetmd.cpp:463|virtual void CNetMD::run()|libnetmd action returned with error:  -4
2023-09-16 18:05:42.999|INFO|cnetmd.cpp:522|void CNetMD::procEnded(bool, int)|
2023-09-16 18:05:41|CRITICAL|bulkTransfer():584: USB transfer error while transffering 1048576 bytes: Operation timed out
2023-09-16 18:05:41|CRITICAL|sendTrack():1029: Error transferring track packets!
2023-09-16 18:05:41|CRITICAL|sendAudioTrack():1477: sendTrack() failed!
2023-09-16 18:05:41|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:05:42|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:05:42|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:05:42.999|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: false ret: -4

2023-09-16 18:09:00.874|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 18:09:00.875|INFO|cjacktheripper.cpp:497|QString CJackTheRipper::deviceInfo()|Found CD Device  "DiscSoft Virtual 1.0"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.eD41iZ.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.IiEnz6.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.MK4PS5.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.1mCK87.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.jknr51.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.D9unvV.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.9Kub36.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.J3G368.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.8dYkZP.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.NiKKkt.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.6LfbXg.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.3g7Qhf.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.33LQu1.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.931302.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.rKfBAf.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.DsKSYn.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.8U3622.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.J37FWU.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.D7DcdC.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.idcls0.tmp"
2023-09-16 18:09:00.996|DEBUG|helpers.cpp:263|QString tempFileName(const QString &)|"/cd2netmd.grG31T.tmp"
2023-09-16 18:09:00.997|INFO|cjacktheripper.cpp:117|int CJackTheRipper::extractTrack(int, const QString &, bool)|Extract track -1 to /private/var/folders/t6/s177fksd1lg0sl6h_9lty01r0000gn/T/cd2netmd.eD41iZ.tmp ...
2023-09-16 18:09:00.997|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: true ret: 0
2023-09-16 18:14:49.918|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: true ret: 0
2023-09-16 18:14:49.921|INFO|cnetmd.cpp:272|int CNetMD::writeTrack(const CNetMD::NetMDCmd &, const QString &, const QString &)|send track: "DAO All in One!" file: "/private/var/folders/t6/s177fksd1lg0sl6h_9lty01r0000gn/T/cd2netmd.eD41iZ.tmp" to "Sony MZ-NE410/DN430/NF520"
2023-09-16 18:15:06.363|CRITICAL|cnetmd.cpp:463|virtual void CNetMD::run()|libnetmd action returned with error:  -4
2023-09-16 18:15:06.363|INFO|cnetmd.cpp:522|void CNetMD::procEnded(bool, int)|
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:14:50|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:15:05|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:15:05|CRITICAL|sendTrack():1013: Error while secureExchange()!
2023-09-16 18:15:05|CRITICAL|sendAudioTrack():1477: sendTrack() failed!
2023-09-16 18:15:05|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:15:06|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:15:06|CRITICAL|sendCmd():397: libusb_control_transfer failed! No such device (it may have been disconnected)
2023-09-16 18:15:06.363|INFO|mainwindow.cpp:533|void MainWindow::transferFinished(bool, int)|checkBusy: false ret: -4

**Comment by Jo2003** [comment]
Hi, do you have any other NetMD softwaare running while using NetMD Wizard? E.g. Web MiniDisc. How do you connect your NetMD device to your Mac? I have seen issues with fast USB hubs. 

What I see is that the NetMD device gets lost while the program tries to write.

**Comment by DAN13L83** [comment]
Hi, I have used Platinum MD and Web minidisc Pro with no issues. My NetMD is connected to a USB-C dock very similar and same brand as the link at the bottom. I have tried a USB-C to Mini USB directly to the M2 MacBook Air and same issue. My Mac mini intel uses an Anker hub that is similar and doesn't have issues. 

https://www.bestbuy.com/site/satechi-type-c-multi-port-adapter-v2-4k-hdmi-ethernet-usb-c-sd-micro-usb-3-0-macbook-pro-macbook-air-windows-laptops-space-gray/6366587.p?skuId=6366587

**Comment by Jo2003** [comment]
Again: Is there any other netmd app working while you try NetMD Wizard? 
Have you set the log level to debug for the traces? 

I don't have any M based Mac. So my possibilities to check are limited.

**Comment by DAN13L83** [comment]
No other apps running. I have set the log to debug. I think the issue was the hub. I have a Usb-c to mini usb that I connected to directly to the macbook air. It worked but that is weird that I tried that before and it didn't work before.

**Comment by Jo2003** [comment]
I had some similar issues on Linux and the only thing which helped, was a very cheap USB2 hub. I think there are some protocol issues between USB3x and USB1.1 which the USB2 hub solves.

**Comment by DAN13L83** [comment]
I gave up, I tried multiple hub configurations. I wanted to dock everything in one usb-c port. At least your advice in the last comment helped. I used the second USB-C port and used a USB-c adapter to a USB 2.0 hub. My goal was to have one of the NetMD always plugged in with a USB switch so that I don't wear out the USB mini port from constant removal. I hardly ever use my MacBook as a Macbook anyways so disconnecting two usb-c plugs once a month should not be inconvenient.

It's working fine now. Thanks for your help.
